crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+pub struct TrustpayAuthUpdateResponse {
+    pub access_token: Option<String>,
+    pub token_type: Option<String>,
+    pub expires_in: Option<i64>,
+    #[serde(rename = "ResultInfo")]
+    pub result_info: ResultInfo,
+}
+
+#[derive(Default, Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct TrustpayAccessTokenErrorResponse {
+    pub result_info: ResultInfo,
+}
+
+impl<F, T> TryFrom<types::ResponseRouterData<F, TrustpayAuthUpdateResponse, T, types::AccessToken>>
+    for types::RouterData<F, T, types::AccessToken>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, TrustpayAuthUpdateResponse, T, types::AccessToken>,
+    ) -> Result<Self, Self::Error> {
+        match (item.response.access_token, item.response.expires_in) {
+            (Some(access_token), Some(expires_in)) => Ok(Self {
+                response: Ok(types::AccessToken {
+                    token: access_token,
+                    expires: expires_in,
+                }),
+                ..item.data
+            }),
+            _ => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: item.response.result_info.result_code.to_string(),
+                    message: item
+                        .response
+                        .result_info
+                        .additional_info
+                        .unwrap_or_default(),
+                    reason: None,
+                    status_code: item.http_code,
+                }),
+                ..item.data
+            }),
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TrustpayRefundRequestCards {
+    instance_id: String,
+    amount: String,
+    currency: String,
+    reference: String,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TrustpayRefundRequestBankRedirect {
+    pub merchant_identification: MerchantIdentification,
+    pub payment_information: BankPaymentInformation,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct TrustpayRefundRequestWrapper {
+    pub card_refunds: Option<TrustpayRefundRequestCards>,
+    pub bank_refunds: Option<TrustpayRefundRequestBankRedirect>,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for TrustpayRefundRequestWrapper {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        match item.payment_method {
+            storage_models::enums::PaymentMethod::BankRedirect => {
+                let auth: TrustpayAuthType = (&item.connector_auth_type)
+                    .try_into()
+                    .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+                let amount =
+                    utils::to_currency_base_unit(item.request.amount, item.request.currency)?
```

Comment:
assume utils::to_currency_base_unit will return f64 and make the necessary change